src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -555,6 +555,8 @@ private IAIState lookForRequests()
                     final List<ItemStack> niceToHave = itemsNiceToHave();
                     final List<ItemStack> contained = InventoryUtils.getContainedFromItemHandler(firstDeliverableRequest.getDeliveries(), worker.getItemHandlerCitizen());
                     final List<ItemStack> missing = InventoryUtils.getMissingFromItemHandler(firstDeliverableRequest.getDeliveries(), worker.getItemHandlerCitizen());
+                    missing.addAll(firstDeliverableRequest.getDeliveries());
```

Comment:
hm dont really get the purpose of this, isnt the list of missing items from the requests added in the line above this?

Reply:
oh yeah, that's the one I only added for debug purposes. I'd like to remove the "missing" items search and actually transfer all items.
(This is the one I want to discuss with orion

Reply:
Well it depends a bit on how you see requests to be honest. In my opinion this does not make any sense. If I remember correctly that method also returns partial stack so even if you request 64 logs while having 32 that would return the required 32 as missing.

I see requests as follows: I am in need of x of item y. Bring me please. If I already have z of y in my inventory I would only need x-z to complete that requests.

Which is what this defies. 

How ever you could also see requests like this: I am missing x of item y. Bring me please. Then this behavior would be correct since you do not care about what he has.

That said. Even our old system worked like the first option, and so does the entire rs. It is why we check buildings and everything first. Before actually performing the request. If we now say we do not want that behavior anymore we would need to rewrite a hole lot of systems, subsystems and workers to make option 2 even functional.

So no I do not think that this makes sense.

Reply:
The problem is that we can request maximum 1 stack. But if we need 128 or 512 of x and have 64 in the inventory he will never pick more up.

Reply:
So either we extend the system to requesting bigger quantities, or we let him pick up the exact quantity he requested, since on request he should've checked already how much he needs compared to how much he has (and not on pickup)

Reply:
On the other hand if you do not care about how much he requests VS how much he already has. This would force him him to pick up every damn stack that was delivered, and the line above it would be become redundant.

Reply:
On the other hand if you do not care about how much he requests VS how much he already has. This would force him him to pick up every damn stack that was delivered, and the line above it would be become redundant.

Reply:
There are multiple ways to fix this 
- Allow requests of more than stacksize but only deliver max stacksize at once (causing a new request on delivery)
- Allow and support requests of more than stacksize
=> And/Or as a workaround until we do that, consider that a worker knows how much he requested on purpose and let him pick up the entire delivered stack (Since I coded most of the AIs I am pretty sure that this won't break anything).

Reply:
The rs has no internal limit to how big a stack that it requests can be. Might be that the requestable however only accepts itemstacks do then you would need to expose the stack limit as well